### PR TITLE
Adding separate route and page for Blog tags

### PR DESCRIPTION
**Pull request by Pallavi-PH**
In this PR we are 

1. Adding separate route and page for Blog tags
screencapture-localhost-3000-blog-tag-OpenEBS-2021-08-12-19_01_37

2. Displaying blog tags at the footer of the blog page
Screenshot 2021-08-12 at 6 54 06 PM

3. Making tags clickable on the blog page, blog slider, and author page and redirecting them to the respective tag page

### CODE REVIEW

File: website/src/pages/Blog/Tag/index.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  const fetchBlogs = async () => {
+    const { default: blogs } = await import(`../../../posts.json`);
+    const filteredData = blogs.filter(
+      (blog: blog) => blog.tags.find((tag: string) => tag.toLowerCase() === selectedTag?.toLowerCase())
+    );
+    setFilteredData(filteredData);
+  };
+
+  useEffect(() => {
+    fetchBlogs();
```
Comment (IsAmrish) [inline website/src/pages/Blog/Tag/index.tsx:61]: If you put the definition of `fetchBlogs` inside the useEffect , you will not have to `disable the linting error` :)

File: website/src/pages/Blog/Tag/index.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        page={page}
+        onChange={(_event, val) => changePage(val)}
+        className={classes.pagination}
+      />
+    );
+  };
+
+  return (
+    <>
+     <Metadata title={SeoJson.pages.blog.title} description={SeoJson.pages.blog.description} url={`${currentOrigin}${SeoJson.pages.blog.url}`} image={`${currentOrigin}${SeoJson.pages.blog.image}`} isPost={false} type={METADATA_TYPES.SERIES}  />
```
Comment (IsAmrish) [inline website/src/pages/Blog/Tag/index.tsx:88]: Metadata value should be related to tags, not individual blog :)

File: website/src/pages/Blog/Tag/index.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  ]);
+  const selectedTag = useTag();
+  const itemsPerPage = 6;
+  const [page, setPage] = React.useState<number>(1);
+  const history = useHistory();
+
+  const fetchBlogs = async () => {
+    const { default: blogs } = await import(`../../../posts.json`);
+    const filteredData = blogs.filter(
+      (blog: blog) => blog.tags.find((tag: string) => tag.toLowerCase() === selectedTag?.toLowerCase())
```
Comment (IsAmrish) [inline website/src/pages/Blog/Tag/index.tsx:55]: blog: blog ==> please change the interface name

File: website/src/pages/Blog/Tag/index.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  excerpt: string;
+  author_info: string;
+  date: string;
+  tags: Array<string>;
+  content: string;
+  id: number;
+  slug: string; 
+}
+
+const Blog: React.FC = () => {
```
Comment (IsAmrish) [inline website/src/pages/Blog/Tag/index.tsx:32]:
```suggestion
const Tag: React.FC = () => {
```

File: website/src/pages/Blog/Tag/styles.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  },
+  blogsWrapper: {
+    justifyContent: 'space-between',
+    [theme.breakpoints.down("sm")]: {
+      justifyContent: 'center',
+    },
+  },
+  cardSize: {
+    maxWidth: "480px !important",
+    paddingTop: theme.spacing(4),
```
Comment (IsAmrish) [inline website/src/pages/Blog/Tag/styles.ts:62]: There is some change in css (ref: https://github.com/openebs/website/pull/210)

File: website/src/pages/Blog/Tag/index.tsx (hunk 1 above)
Comment (Pallavi-PH) [reply website/src/pages/Blog/Tag/index.tsx:61]: It is disabled. Please check line no. 62

File: website/src/pages/Blog/Tag/styles.ts (hunk 5 above)
Comment (Pallavi-PH) [reply website/src/pages/Blog/Tag/styles.ts:62]: This will auto added after rebase :)

File: website/src/pages/Blog/Tag/index.tsx (hunk 1 above)
Comment (IsAmrish) [reply website/src/pages/Blog/Tag/index.tsx:61]: yes, it is!

Comment (IsAmrish) [reply website/src/pages/Blog/Tag/index.tsx:61]: What am I saying is, if you put the definition of `fetchBlogs` inside the `useEffect`, you will not have to disable the lining error ... means you can remove `// eslint-disable-next-line react-hooks/exhaustive-deps`

File: website/src/pages/Blog/Tag/styles.ts (hunk 5 above)
Comment (IsAmrish) [reply website/src/pages/Blog/Tag/styles.ts:62]: ref: https://github.com/openebs/website/pull/210/files#diff-63af27bc462eb8b5d718572232baac7a03137a42bd14704b72430a21f09c2ecaR110

File: website/src/pages/Blog/Tag/index.tsx (hunk 3 above)
Comment (satyapriyamishra222) [reply website/src/pages/Blog/Tag/index.tsx]: blog: BlogItem

File: website/src/hooks/extractBlogPath.ts [hunk 6]
```diff
@@ -12,4 +12,10 @@ const useAuthorName = () => {
   return location.pathname.split("/blog/author/").pop();
 };
 
-export { useBlogName, useAuthorName };
\ No newline at end of file
+// custom hook for getting blog tag from the URL
+const useTag = () => {
+  const location = useLocation();
+  return location.pathname.split("/blog/tag/").pop();
```
Comment (satyapriyamishra222) [inline website/src/hooks/extractBlogPath.ts:18]: What exactly is this doing ? Do we need this kind of implementation for this.

Comment (Pallavi-PH) [reply website/src/hooks/extractBlogPath.ts:18]: We need this to get the tag name from the URL.  In this `extractBlogPath` hooks, we have similar functions used in blogs to get author name and blog name from URL. So added it here for grouping of functions

File: website/src/pages/Blog/Tag/index.tsx (hunk 2 above)
Comment (Pallavi-PH) [reply website/src/pages/Blog/Tag/index.tsx]: This is also list of blogs but filtered with tag. So we can keep the same title and description. Changed URL

**Review by satyapriyamishra222 — APPROVED** [review]

File: website/src/pages/Blog/Tag/index.tsx [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+  ]);
+  const selectedTag = useTag();
+  const itemsPerPage = 6;
+  const [page, setPage] = React.useState<number>(1);
+  const history = useHistory();
+
+  
+
+  const handleTagSelect = (tag: string) => {
+    if(selectedTag !== tag){
```
Comment (rakeshPRaghu) [inline website/src/pages/Blog/Tag/index.tsx:55]: the check should be something like this 

```suggestion
    if(selectedTag !== tag.toLowerCase().replace(/[^\w ]+/g,'').replace(/ +/g,'-')){
```

**Review by IsAmrish — APPROVED** [review]

**Review by AVRahul — APPROVED** [review]
LGTM, Thanks @Pallavi-PH

**Pull request merged**